pkg/stanza/fileconsumer/internal/tracker/tracker_other.go

```diff
@@ -0,0 +1,21 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build !windows
+
+package tracker // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/tracker"
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fileset"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader"
+)
+
+// On non-windows platforms, we keep files open between poll cycles so that we can detect
+// and read "lost" files, which have been moved out of the matching pattern.
+func (t *Tracker) EndConsume() {
+	t.ClosePreviousFiles()
+
+	// m.currentPollFiles -> m.previousPollFiles
+	t.previousPollFiles = t.currentPollFiles
+	t.currentPollFiles = fileset.New[*reader.Reader](t.maxBatchFiles)
```

Comment:
This is a change in behavior since we weren't doing it before. Let's just keep this PR to a pure refactor. Please propose the change in a separate PR if you think it's necessary.
```suggestion
```

Reply:
@djaglowski I think this is necessary.
Because we need to reset `currentPollFiles` after every `consume()` call. This makes sure that the fileset is ready for a new cycle.
Without clearing/resetting, we will find same set of readers in `previousPollFiles` and `currentPollFiles`.

Albeit we can do this differently, at the beginning of each `consume()`. But we need to do it.

> This is a change in behavior since we weren't doing it before.

We are currently doing it [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/fileconsumer/file.go#L218).
Now that the responsibility to handle filesets has been transferred to `tracker`, I've moved that piece of code in `tracker` interface.

Let me know what do you think.

Reply:
Ah, makes sense. Just need to correct it to `t.maxBatchFiles / 2` then.

Reply:
Hmm, I think we crossed wires on max concurrent files and max batch files at some point because we are splitting both in half at various points in the code. I don't think we ever need to divide max batch files in half because it's already half of the max concurrent. Can you sanity check me on this?

Reply:
Oh yeah. that's correct. 
`t.maxBatchFiles` itself is half of `maxConcurrentFiles`. I've changed the code where we made that mistake.
PTAL and I guess we can merge if it looks good to you.